### DISCUSSION TITLE
how to bind to NUMBER

### DISCUSSION THREAD

**Discussion by begemot63** [godror/godror#325] [General]
Hi, 
I use in golang type:   type ResourceType int
and I need to bind int to the column NUMBER(4.0) 
I got error:
Handling uncaught error: type=*data.DBException, error=error executing query: 2. arg bindVarTypeSwitch(model.ResourceType): unknown type

My code uses custom type ResourceType:

``
type ResourceType int
``

 The driver cannot covert custom type ResourceType to integer , so I have to do it implicitly, for example instead of
  dao.executeListQuery(sqlSelectTaskByEntityKey, resourceType)
I need to use :
  dao.executeListQuery(sqlSelectTaskByEntityKey, int(resourceType))

Is any better way to handle custom type ResourceType to allow driver to handle the  resourceType variable as a integer in this call ?

**Comment by tgulacsi** [comment]
No, not yet.
We could look into the type to see the "underlying type", but it's hard to not gain complexity in the driver.
I'm open for suggestions!

**Comment by begemot63** [comment]
Thanks Tamás,

My code uses custom type ContractStatus:

type ContractStatus int

The driver cannot convert custom type *ContractStatus* to integer , so I
have to do it implicitly, for example instead of

err := dao.ExecContext(
    sqlUpdateContractStatusAndMessage, status, nullErrMsg, key.ID, key.Version)

 I need to use :

err := dao.ExecContext(
    sqlUpdateContractStatusAndMessage, int(status), nullErrMsg,
key.ID, key.Version)

Is there any better way to handle custom type ContractStatus to allow the
driver to handle the status variable as an integer in this call ?

**Comment by tgulacsi** [comment]
PTAL 89780a1
I've added some reflection for such cases - it's inefficient, but works.
